### PR TITLE
Bump minimum Apple OS versions

### LINKED ISSUE #91736: Bump iOS/tvOS minimum supported version to 12.2
Xcode 15 drops support for targeting iOS 11.0 (our current baseline) and bumping to 12.2 allows us to rely on the Swift libraries shipped with the OS.

Also bump the MacCatalyst versions (`apple-iosXY-macabi`) to 15.0 since that is the lowest which works on macOS 12

### PR DESCRIPTION
**Pull request by akoeplinger**
As defined in the ".NET 9 - Supported OS versions" document: https://github.com/dotnet/core/blob/main/release-notes/9.0/supported-os.md

- macOS 12.0
- iOS/iOSSimulator 12.2
- tvOS/tvOSSimulator 12.2
- MacCatalyst 15.0 (== macOS 12.0)

Fixes https://github.com/dotnet/runtime/issues/91736

### CODE REVIEW

**Review by filipnavara — APPROVED** [review]

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/MachObjectWriter.cs [hunk 1]
```diff
@@ -244,15 +244,12 @@ private protected override void EmitObjectFile(string objectFilePath)
...
                     break;
 
                 case TargetOS.MacCatalyst:
                     buildVersion.Platform = PLATFORM_MACCATALYST;
-                    buildVersion.MinimumPlatformVersion = _cpuType switch
-                    {
-                        CPU_TYPE_X86_64 => 0x0D_05_00u, // 13.5.0
-                        _ => 0x0E_02_00u, // 14.2.0
+                    buildVersion.MinimumPlatformVersion = 0x0F_02_00; // 15.0.0
                     };
```
Comment (filipnavara) [inline src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/MachObjectWriter.cs:253]:
```suggestion
```

**Comment by filipnavara** [comment]
Looks like pal_signverify.c will need to silence a couple of deprecation warnings in the DSA code (`SecTransformSetAttribute` / `SecTransformExecute`). We already have a pattern for it elsewhere in the code:
https://github.com/dotnet/runtime/blob/a3dc1337a17a3de6aefe76e465f85ebf5ba6ae5e/src/native/libs/System.Security.Cryptography.Native.Apple/pal_keychain_macos.c#L19-L22
There's no replacement for these APIs for this specific security algorithm, so the only other alternative is to drop the support entirely which is out of scope for this PR.

**Comment by akoeplinger** [comment]
Thanks, will add the suppression.

We're also seeing this is the maccatalyst build:
> src/native/libs/System.Security.Cryptography.Native.Apple/pal_ssl.c(466,34): error G61EDB71E: (NETCORE_ENGINEERING_TELEMETRY=Build) 'SecTrustGetCertificateAtIndex' is deprecated: first deprecated in macCatalyst 15.0

**Comment by filipnavara** [comment]
The `SecTrustGetCertificateAtIndex` deprecation would need a new code path to use `SecTrustCopyCertificateChain`. To make that efficient we would likely need to rewrite the managed -> native interface. I guess it makes sense to suppress the deprecation and file an issue for a follow up work. I guess either I, or perhaps @vcsjones, can take a crack at it.

**Comment by akoeplinger** [comment]
Thanks, pushed the suppression and filed https://github.com/dotnet/runtime/issues/101349

I also got an error when building locally with Xcode 15 from the new linker about -bind_at_load being deprecated for macOS (looks like it checks the min OS target since we didn't hit this before) so I also added a check to not pass that flag for the new linker.

**Comment by vcsjones** [comment]
> I guess either I, or perhaps @vcsjones, can take a crack at it.

There are a number of deprecation suppressions we can tackle with the iOS version bump. I already have a branch to start this, so I can include this.

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesGcm.macOS.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@ public sealed partial class AesGcm
     {
         private FixedMemoryKeyBox _keyBox;
 
-        // CryptoKit added AES.GCM in macOS 10.15, which is our minimum target for macOS.
+        // CryptoKit added AES.GCM in macOS 10.15, which is higher than our minimum target for macOS.
```
Comment (ivanpovazan) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesGcm.macOS.cs:14]: nit: Shouldn't this be "lower than our minimum target.."?

**Review by ivanpovazan — APPROVED** [review]
LGTM
Just a small comment: I noticed that the table in https://github.com/dotnet/core/blob/main/release-notes/9.0/supported-os.md is not in sync with these changes e.g., MacCatalyst version is specified as `12.0+`

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesGcm.macOS.cs (hunk 2 above)
Comment (akoeplinger) [reply src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesGcm.macOS.cs:14]: Ah right, I meant to say that our minimum target is higher than 10.15, will reword it :)

**Comment by akoeplinger** [comment]
> ... is not in sync with these changes e.g., MacCatalyst version is specified as 12.0+

Yeah this was accidentally overwritten: https://github.com/dotnet/core/pull/9279

**Pull request merged**